packages/devtools_app/lib/src/inspector/inspector_controller.dart

```diff
@@ -915,4 +922,17 @@ class InspectorController extends DisposableController
       onLayoutExplorerSupported,
     );
   }
+
+  Stream<String> _collectValidInspectorRefs(
```

Comment:
is there anyway we could create a set of all these ids at the point that we compute the tree? That way we wouldn't have to traverse the tree twice and we would have have quick lookup time for seeing if the set contains one of the error-causing inspectorRefs

@jacob314 for some input here.

Reply:
this is way too slow because you are awaiting fetching children one at a time possibly for the entire tree.
The computed tree already includes the ids and has a set similar to this.
The challenge is that will only be ids from the summary tree so make sure you are only needing to compare against a node id from the summary tree.
For the detail tree we don't load the whole tree as it is huge.

Reply:
Ah, didn't realise this code was used in both trees. I can guard it with a check of `isSummaryTree` (my understanding is that the widgets are all in the summary tree and this should be fine?).

> The computed tree already includes the ids and has a set similar to this.

Where is the computed tree? It looked like these calls to `children` were all bailing out early because they already had the data, but I don't know if that's guaranteed or not (or if that's dependant on the `await` I added to the `recomputeTreeRoot` call?).

Reply:
How will we handle errors for widgets that are not in the summary tree? I think this could be a pretty common occurrence, right @jacob314?

Reply:
Ah, I wasn't sure if that was possible. Right now IDs that aren't in the summary tree will just be pruned out. We'd need to be able to tell which ones are still valid that aren't in the summary tree to support this.

Is there an error I can generate that has the deep link that wouldn't show up in the summary tree?